src/python/pants/backend/awslambda/python/target_types.py

```diff
@@ -47,11 +64,49 @@ class ResolvePythonAwsHandlerRequest:
     field: PythonAwsLambdaHandlerField
 
 
-@rule
+@rule(desc="Determining the handler for a `python_awslambda` target")
 async def resolve_python_aws_handler(
     request: ResolvePythonAwsHandlerRequest,
 ) -> ResolvedPythonAwsHandler:
-    return ResolvedPythonAwsHandler(request.field.value)
+    handler_val = request.field.value
+    field_alias = request.field.alias
+    address = request.field.address
+    path, _, func = handler_val.partition(":")
+
+    # If it's already a module, simply use that. Otherwise, convert the file name into a module
+    # path.
+    if not path.endswith(".py"):
+        return ResolvedPythonAwsHandler(handler_val)
```

Comment:
`py` or `pyi`?

Reply:
I wasn't sure - I can't really imagine an entry point being a type stub. Type stubs aren't supposed to be runnable. I figure that we can add it if some user explains to us a use case we aren't aware of.

Reply:
The type stub isn't runnable, but it declares the existence of something that is runnable? Not a blocker for sure.

Reply:
That's true, that we would convert the `.pyi` into a module with the `.pyi` stripped. Still, weird, and easy workaround + easy to change.